File: packages/python/langchain_nuggets/tools/kya/get_agent_trust_score.py
"""Get agent trust score tool."""
from __future__ import annotations
from urllib.parse import quote

import json
from typing import Optional, Type

from langchain_core.callbacks import AsyncCallbackManagerForToolRun, CallbackManagerForToolRun
from pydantic import BaseModel, Field

from langchain_nuggets.tools.base import NuggetsBaseTool


class GetAgentTrustScoreInput(BaseModel):
    agentId: str = Field(description="The agent ID or DID to get the trust score for")


class GetAgentTrustScore(NuggetsBaseTool):
    name: str = "get_agent_trust_score"
    description: str = "Get the trust score and provenance signals for an AI agent. Returns a score (0-1) based on verified signals: GitHub account verification, social profile verification, and registration age. Higher scores indicate more trustworthy agents with stronger developer provenance."
    args_schema: Type[BaseModel] = GetAgentTrustScoreInput

    def _run(self, agentId: str, run_manager: Optional[CallbackManagerForToolRun] = None) -> str:
        result = self.client.get(f"/kya/agents/{quote(agentId, safe="")}/trust-score")
        return json.dumps(result)

    async def _arun(self, agentId: str, run_manager: Optional[AsyncCallbackManagerForToolRun] = None) -> str:
        result = await self.client.aget(f"/kya/agents/{quote(agentId, safe="")}/trust-score")
        return json.dumps(result)

File: packages/js/src/tools/kya/get-agent-trust-score.ts
import { z } from "zod";
import { NuggetsBaseTool } from "../base.js";
import type { AgentTrustScore } from "../../types.js";

const schema = z.object({
  agentId: z.string().describe("The agent ID or DID to get the trust score for"),
});

export class GetAgentTrustScore extends NuggetsBaseTool {
  name = "get_agent_trust_score";
  description = "Get the trust score and provenance signals for an AI agent. Returns a score (0-1) based on verified signals: GitHub account verification, social profile verification, and registration age. Higher scores indicate more trustworthy agents with stronger developer provenance.";
  schema = schema;

  async _call({ agentId }: z.output<typeof schema>): Promise<string> {
    const score = await this.client.get<AgentTrustScore>(`/kya/agents/${encodeURIComponent(agentId)}/trust-score`);
    return JSON.stringify(score);
  }
}
